tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot

```diff
@@ -123,28 +134,33 @@ Spawn Notebook With Arguments
    ...              By creating a dictionary beforehand, e.g. &{test-dict}  Create Dictionary  name=robot  password=secret
    [Arguments]  ${retries}=1  ${image}=s2i-generic-data-science-notebook  ${size}=Small  ${spawner_timeout}=600 seconds  &{envs}
    FOR  ${index}  IN RANGE  0  1+${retries}
-      Select Notebook Image  ${image}
-      Select Container Size  ${size}
-      IF  &{envs}
-         Remove All Spawner Environment Variables
-         FOR  ${key}  ${value}  IN  &{envs}[envs]
-            Sleep  1
-            Add Spawner Environment Variable  ${key}  ${value}
+      ${spawner_ready} =    JupyterHub Spawner Is Ready
+      IF  ${spawner_ready}==True
+         Select Notebook Image  ${image}
+         Select Container Size  ${size}
+         IF  &{envs}
+            Remove All Spawner Environment Variables
+            FOR  ${key}  ${value}  IN  &{envs}[envs]
+               Sleep  1
+               Add Spawner Environment Variable  ${key}  ${value}
+            END
          END
-      END
-      ${version-check} =  Is RHODS Version Greater Or Equal Than  1.5.0
-      IF  ${version-check}==True
-         Click Button  Start Server
-         Wait Until Page Contains  Starting server
+         ${version-check} =  Is RHODS Version Greater Or Equal Than  1.5.0
+         IF  ${version-check}==True
+            Click Button  Start Server
+            Wait Until Page Contains  Starting server
+         ELSE
+            Click Button  Start server
+            Wait Until Page Contains  Your server is starting up
+         END
```

Comment:
do you think it will be good idea to move line 148~155 in some keyword and call that keyword? I know it will not make much difference just for the easy readability

Reply:
Yeah, that might be better